### ISSUE TITLE
Deepspeed + Custom LR Scheduler + accelerate.accumulate does not enforce gradient accumulation

### ISSUE THREAD

**Issue by yuvalkirstain** [huggingface/accelerate#995]
### System Info

```Shell
- `Accelerate` version: 0.15.0
- Platform: Linux-4.15.0-65-generic-x86_64-with-glibc2.27
- Python version: 3.10.8
- Numpy version: 1.23.5
- PyTorch version (GPU?): 2.0.0.dev20230105 (True)
- `Accelerate` default config:
	- compute_environment: LOCAL_MACHINE
	- distributed_type: DEEPSPEED
	- mixed_precision: no
	- use_cpu: False
	- dynamo_backend: NO
	- num_processes: 4
	- machine_rank: 0
	- num_machines: 1
	- gpu_ids: None
	- main_process_ip: None
	- main_process_port: None
	- rdzv_backend: static
	- same_network: True
	- main_training_function: main
	- deepspeed_config: {'gradient_accumulation_steps': 4, 'offload_optimizer_device': 'none', 'offload_param_device': 'none', 'zero3_init_flag': False, 'zero_stage': 2}
	- fsdp_config: {}
	- megatron_lm_config: {}
	- downcast_bf16: no
	- tpu_name: None
	- tpu_zone: None
	- command_file: None
	- commands: None
```

### Information

- [ ] The official example scripts
- [X] My own modified scripts

### Tasks

- [ ] One of the scripts in the examples/ folder of Accelerate or an officially supported `no_trainer` script in the `examples` folder of the `transformers` repo (such as `run_no_trainer_glue.py`)
- [X] My own task or dataset (give details below)

### Reproduction

Issue: the LR scheduler changes the LR after each `step` rather than after `gradient_accumulation_steps`. Is it the expected behaviour?

Steps to reproduce:
1. Choose any script that uses accelerate with deepspeed + custom LR scheduler + `with accelerator.accumulate(model):` 
The script should contain a piece of code that instantiates a custom LR scheduler:
```python
  lr_scheduler = get_scheduler(
      cfg.lr_scheduler.type,
      optimizer=optimizer,
      num_warmup_steps=total_warmup_steps,
      num_training_steps=total_training_steps,
  )
```
and uses accelerator.accumulate(model):
```python
            with accelerator.accumulate(model):
                outputs = model(**batch)
                loss = outputs.loss
                accelerator.backward(loss)
                optimizer.step()
                lr_scheduler.step()
                optimizer.zero_grad()
```
2. Make sure that we use gradient accumulation > 1.
3. Print the LR after each step 
```python
            with accelerator.accumulate(model):
                outputs = model(**batch)
                loss = outputs.loss
                accelerator.backward(loss)
                optimizer.step()
                lr_scheduler.step()
                logger.info(lr_scheduler.get_last_lr()[0]) # Added this line
                optimizer.zero_grad()
```
4. I expected the context manager of `accelerator.accumulate` to change LR only after `gradient_accumulation_steps`, but it changes it every step. 

### Expected behavior

I expected the LR scheduler to get updated after each backward step. If we should update it after each step, then that's OK (but these intermediate updates will have no effect).

**Comment by pacman100** [comment]
Hello, this issue is irrespective of using DeepSpeed, please refer #963

**Comment by yuvalkirstain** [comment]
Oh cool. Is fixing the bug anything more than adding:
```python
if self.gradient_state.sync_gradients:
```
at the start of the `step` method in acceleratedScheduler? 
If it is enough, I can create a PR :)

**Comment by yuvalkirstain** [comment]
Oh great, commented in #963 so I'll close this issue. Thank you!
